ui/js/models/repository.js

```diff
@@ -141,13 +141,28 @@ treeherder.factory('ThRepositoryModel', [
                         });
 
                         _.each(data, addRepoAsUnwatched);
+
+                        // This needs to be done before `setCurrent` because
+                        // `setCurrent` overwrites the entire listing
+                        // with only the default repo
+                        if (options.watchRepos) {
+                            var storedWatched = loadWatchedRepos();
+                        }
+
                         if (options.name) {
                             setCurrent(options.name);
                         }
                         if (options.watchRepos) {
-                            var storedWatched = loadWatchedRepos();
-                            if (_.isArray(storedWatched) &&
-                                _.contains(storedWatched, options.name)) {
+                            if (_.isArray(storedWatched)) {
+
+                                // To get the current repo to display first, we must
```

Comment:
While this does seem to work, I don't think we can/should rely on this order of creation in the object.  I think the best thing to do is to convert ``watchedRepos`` to an array instead of an object.